### ISSUE TITLE
Automatically test manual / docstring examples

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/nyxt#1580]
Would be great, but does not seem obvious to do.  Maybe some Common Lisp library out there to help?

**Comment by aadcg** [comment]
@aartaka I don't think 2ef4da0 guarantees that the code actually works right?

**Comment by aartaka** [comment]
It does not, yes. But it lays the ground for 573442a9c9c66bd576b1778f8ff675d8d1bef701, which _should_ guarantee the code is working. I've not tested it, though. 

UPD: I ran `(asdf:test-system "nyxt/application/tests")` and it succeeded with only one test (about `NYXT/TESTS/EXECUTABLE::CONFIG-FAILS-AND-BROWSER-RESTARTS`, not related to manual examples) failing!

**Comment by aartaka** [comment]
But building Nyxt just to run a small set of tests is an overkill. Maybe we can create dummy browser, set `*run-from-repl-p*` to T, and catch all the errors instead of checking the error code in shell?

**Comment by aadcg** [comment]
Oh I see, I missed 573442a.

**Comment by aadcg** [comment]
I think we should test 573442a a bit more, and also keep in mind that it covers manual section but not docstring examples.

**Comment by aartaka** [comment]
Well, docstring examples I wouldn't hope to test, as it's an AGI-complete problem involving natural language docstring parsing :)

So, two ways of solving it that I can see are:
- Sticking to an easily parseable syntax in docstrings, or
- Using a mostly-correct parser to slurp code snippets from docstrings.

The former is restricting us and making our docstrings less readable anywhere beyond Nyxt. The latter is likely to go berserk in 5% of cases where the syntax is irregular and is thus not suitable for testing.

I don't like either of those. But I do recognize the problem of docstring testing.

**Comment by aadcg** [comment]
Generally speaking, I don't see why there shouldn't be a template to insert a code snippet. I believe we already have an informal one which goes by "Example:".

**Comment by aadcg** [comment]
I don't think this is a priority at all anyway (both with regards to the manual and docstrings).
